### PR TITLE
Add frontend test infrastructure and unit tests

### PR DESCRIPTION
**Pull request by nicolai-rhesis**
## Purpose

Establish a solid frontend testing foundation with working coverage collection, unit tests for utility functions, hooks, and shared components.

## What Changed

- **Fixed CI coverage collection**: switched to V8 coverage provider to resolve the `inflight`/`test-exclude` compatibility issue; added coverage summary output to GitHub Actions
- **Set up MSW (Mock Service Worker)**: added request handlers and server for API mocking in tests
- **Added 472 tests across 34 test suites** covering:
  - Utility functions: url-resolver, validateEnv, onboarding-service, span-icon-mapper, format-duration, validation, url, status-colors, entity-helpers, comment-utils, odata-filter, git-utils, entity-error-handler, date-utils, test-result-status, trace-utils
  - Hooks: useDocumentTitle, useFormChangeDetection, useComments, useTasks
  - Components: DeleteModal, SearchAndFilterBar, ErrorBoundary, NotificationContext
  - Constants: knowledge, paths, test-types
- **Enabled coverage thresholds** as a ratchet to prevent regression:
  - Statements: 6.4%
  - Branches: 53%
  - Functions: 17%
  - Lines: 6.4%

## Testing

```bash
cd apps/frontend && npm run test:ci
```

All 472 tests pass. Coverage thresholds enforced in CI.

---

> [!NOTE]
> **Medium Risk**
> Moderate risk due to broad UI refactors (key changes across many lists) and new CI coverage thresholds that can fail builds; runtime behavior should be unchanged but rendering/selection state could shift in edge cases.
> 
> **Overview**
> Enables frontend CI test coverage by switching Jest to the `v8` coverage provider, extending `test:ci` to emit `json-summary`/`lcov`, and updating the GitHub Actions workflow to run `npm run test:ci` and post a coverage summary to `$GITHUB_STEP_SUMMARY`.
> 
> Adds MSW-based API mocking (`src/__mocks__/msw`) for Jest tests and introduces a large new unit test suite for shared components, hooks, constants, and utilities.
> 
> Refactors many UI components to use more stable React keys (content-based keys instead of array indices) and makes a small cleanup in `BehaviorCard` by removing an unused caught error variable.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 89e31f5e32d1aebe8c9a4b6fe9cc46e00a47f8ce. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: apps/frontend/eslint.config.mjs [hunk 1]
```diff
@@ -70,7 +70,7 @@ export default [
       // React specific rules
       'react/prop-types': 'off',
       'react/react-in-jsx-scope': 'off',
-      'react/no-array-index-key': 'error',
+      'react/no-array-index-key': 'warn',
```
[the remark this answers is not in the document]
Comment (nicolai-rhesis) [reply apps/frontend/eslint.config.mjs]: Done
